test/PreAMMBatcher.spec.ts

```diff
@@ -171,16 +165,21 @@ describe("PreAMMBatcher", () => {
       testCaseInput.sellOrdersToken1[0],
     );
 
-    await batcher.batchTrade(
-      testCaseInput.sellOrdersToken0[0].encode(),
-      testCaseInput.sellOrdersToken1[0].encode(),
-      { gasLimit: 6000000 },
-    );
-    expect("transferFrom").to.be.calledOnContractWith(token0, [
-      walletTrader1.address,
-      batcher.address,
-      testCaseInput.sellOrdersToken0[0].sellAmount.toString(),
-    ]);
+    await token0.mock.transferFrom.reverts();
```

Comment:
This looks very nice (not needing to encode the method name manually), although I'm also a little sad we are no longer using mock-contract.

Reply:
Yes I was thinking the same thing!

Reply:
> although I'm also a little sad we are no longer using mock-contract.

We are though:
```
    token0 = await waffle.deployMockContract(walletDeployer, ERC20.abi);
    token1 = await waffle.deployMockContract(walletDeployer, ERC20.abi);
```

This uses `@ethereum-waffle/mock-contract` under the hood - we just don't need to include a direct dependency in the `package.json` anymore since it is re-exported through `waffle`.

This change was because of the [`calledOnContractWith` matchers currently don't work with buidler](https://github.com/nomiclabs/buidler/issues/638).

Reply:
I believe he is referring to our own (his) `MockContract` https://github.com/gnosis/mock-contract